src/types.rs

```diff
@@ -245,6 +251,8 @@ pub struct OneExit {
     pub city_name: String,
     pub provider_id: String,
     pub provider_url: String,
+    pub vpn_provider: VpnProvider,
+    pub vpn_provider_url: String
```

Comment:
```suggestion
    pub provider_name: VpnProvider,
    pub provider_homepage_url: String
```

It's a bit weird because we are conflating "provider's info for this server" and "info about the provider itself. But I think that is fine. I think putting `vpn_` prefix is misleading because it sounds like we are talking about 2 different providers, which we aren't. If anything maybe it is best just to put comments to explain what each means.

Reply:
I should also say that for the "provider ID" enum I would just use the name `provider`. `provider_name` would be a human-readable string if we choose to do that.